src/networking.c

```diff
@@ -3848,36 +3847,19 @@ void flushSlavesOutputBuffers(void) {
 
 /* Compute current most restrictive pause type and its end time, aggregated for
  * all pause purposes. */
-static void updateClientPauseTypeAndEndTime(void) {
-    pause_type old_type = server.client_pause_type;
-    pause_type type = CLIENT_PAUSE_OFF;
-    mstime_t end = 0;
-    for (int i = 0; i < NUM_PAUSE_PURPOSES; i++) {
-        pause_event *p = server.client_pause_per_purpose[i];
-        if (p == NULL) {
-            /* Nothing to do. */
-        } else if (p->end < server.mstime) {
-            /* This one expired. */
-            zfree(p);
-            server.client_pause_per_purpose[i] = NULL;
```

Comment:
don't we still wanna reset the pause purpose when its time has expired?
we no longer need to release an allocation, but don't we still wanna reset it to 0?
maybe i'm missing the place where it's done?

Reply:
i have a feeling that "normalizing" it would be nicer and safer.. but no strong feeling either.